src/ui/server/controllers/insurance/export-contract/map-and-save/export-contract-agent/index.ts

```diff
@@ -14,20 +21,49 @@ import { Application, RequestBody, ValidationErrors } from '../../../../../../ty
 const exportContractAgent = async (formBody: RequestBody, application: Application, validationErrors?: ValidationErrors) => {
   try {
     if (hasFormData(formBody)) {
+      const { isUsingAgent } = formBody;
+
       const populatedData = mapSubmittedData(formBody);
 
       let saveResponse;
 
       if (validationErrors) {
-        saveResponse = await save.exportContractAgent(application, populatedData, validationErrors.errorList);
+        saveResponse = await saveAgent.exportContractAgent(application, populatedData, validationErrors.errorList);
       } else {
-        saveResponse = await save.exportContractAgent(application, populatedData);
+        saveResponse = await saveAgent.exportContractAgent(application, populatedData);
       }
 
       if (!saveResponse) {
         return false;
       }
 
+      /**
+       * If AGENT_SERVICE data should be nullified,
+       * Nullify and save all AGENT_CHARGES and AGENT_SERVICE data.
+       */
+      if (shouldNullifyAgentServiceData(isUsingAgent, application.exportContract.agent)) {
```

Comment:
could this go in a helper? will make this file shorter

Reply:
as in the bit which sets service and serviceCharge to null

Reply:
Good shout. Moved into a new function.

Need to do the same for "Loss payee" - one for later.